cygwin/CygwinProcessTable.c

```diff
@@ -0,0 +1,799 @@
+/*
+htop - CygwinProcessTable.c
+(C) 2014 Hisham H. Muhammad
+(C) 2015 Michael McConville
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "cygwin/CygwinProcessTable.h"
+
+#include <stdlib.h>
+#include <dirent.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "Object.h"
+#include "ProcessTable.h"
+#include "XUtils.h"
+#include "cygwin/CygwinProcess.h"
+#include "cygwin/CygwinMachine.h"
+
+
+static FILE* fopenat(openat_arg_t openatArg, const char* pathname, const char* mode) {
+   assert(String_eq(mode, "r")); /* only currently supported mode */
+
+   int fd = Compat_openat(openatArg, pathname, O_RDONLY);
+   if (fd < 0)
+      return NULL;
+
+   FILE* stream = fdopen(fd, mode);
+   if (!stream)
+      close(fd);
+
+   return stream;
+}
```

Comment:
Please avoid the libc `FILE*` APIs and use `read`/`write` with the `fd` directly.

Reply:
> Please avoid the libc `FILE*` APIs and use `read`/`write` with the `fd` directly.

this is copied from the existing Linux support: https://github.com/htop-dev/htop/blob/2978af01a79d1627d7fa90fda0efec9cf774d901/linux/LinuxProcessTable.c#L67-L79

Reply:
I know. And that code is on my todo list of deprecated things that need to be cleaned up at some point …